azurerm/internal/services/storage/tests/resource_arm_hpc_cache_nfs_target_test.go

```diff
@@ -0,0 +1,333 @@
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMHPCCacheNFSTarget_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMHPCCacheNFSTargetExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMHPCCacheNFSTarget_usageModel(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMHPCCacheNFSTargetExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMHPCCacheNFSTarget_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMHPCCacheNFSTargetExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMHPCCacheNFSTarget_namespaceJunction(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_hpc_cache_nfs_target", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMHPCCacheNFSTargetDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMHPCCacheNFSTarget_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMHPCCacheNFSTargetExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMHPCCacheNFSTarget_namespaceJunction(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMHPCCacheNFSTargetExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+			{
+				Config: testAccAzureRMHPCCacheNFSTarget_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMHPCCacheNFSTargetExists(data.ResourceName),
+				),
+			},
+			data.ImportStep(),
+		},
+	})
+}
+
+func TestAccAzureRMHPCCacheNFSTarget_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_hpc_cache_nfs_target", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMHPCCacheNFSTargetDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMHPCCacheNFSTarget_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMHPCCacheNFSTargetExists(data.ResourceName),
+				),
+			},
+			data.RequiresImportErrorStep(testAccAzureRMHPCCacheNFSTarget_requiresImport),
+		},
+	})
+}
+
+func testCheckAzureRMHPCCacheNFSTargetExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("HPC Cache NFS Target not found: %s", resourceName)
+		}
+
+		id, err := parsers.HPCCacheTargetID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		client := acceptance.AzureProvider.Meta().(*clients.Client).Storage.StorageTargetsClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
```

Comment:
minor/not a blocker - ideally these are always at the top of the function they are called in

Reply:
Good to know.